cw-orch-cli/src/commands/action/cosmwasm/store/mod.rs

```diff
@@ -0,0 +1,66 @@
+use crate::{commands::action::CosmosContext, types::keys::seed_phrase_for_id};
+
+use color_eyre::eyre::Context;
+use cw_orch::{daemon::TxSender, prelude::*};
+
+#[derive(Debug, Clone, interactive_clap::InteractiveClap)]
+#[interactive_clap(input_context = CosmosContext)]
+#[interactive_clap(output_context = StoreWasmOutput)]
+/// Execute contract method
+pub struct StoreContractCommands {
+    /// Input path to the wasm
+    wasm_path: crate::types::PathBuf,
+    #[interactive_clap(skip_default_input_arg)]
+    signer: String,
+}
+
+impl StoreContractCommands {
+    fn input_signer(_context: &CosmosContext) -> color_eyre::eyre::Result<Option<String>> {
+        crate::common::select_signer()
+    }
+}
+
+pub struct StoreWasmOutput;
+
+impl StoreWasmOutput {
+    fn from_previous_context(
+        previous_context: CosmosContext,
+        scope:&<StoreContractCommands as interactive_clap::ToInteractiveClapContextScope>::InteractiveClapContextScope,
+    ) -> color_eyre::eyre::Result<Self> {
+        let chain = previous_context.chain;
+        let wasm_byte_code = std::fs::read(&scope.wasm_path).wrap_err(format!(
+            "Failed to open or read the file: {}",
+            scope.wasm_path.0.display()
+        ))?;
+
+        let seed = seed_phrase_for_id(&scope.signer)?;
+        let daemon = chain.daemon(seed)?;
+
+        let upload_msg = cosmrs::cosmwasm::MsgStoreCode {
+            sender: daemon.sender().account_id(),
+            wasm_byte_code,
+            instantiate_permission: None,
+        };
+        let resp = daemon
+            .rt_handle
+            .block_on(daemon.sender().commit_tx(vec![upload_msg], None))?;
+        let code_id = resp.uploaded_code_id().unwrap();
+        println!("code_id: {code_id}");
+
+        Ok(StoreWasmOutput)
+    }
+}
+
+// TODO: the dream here to use Uploadable instead
+// fn uploadable_from_path(wasm_path: WasmPath) -> impl Uploadable {
+//     struct Placeholder {
+//         wasm_path: WasmPath,
+//     }
+//     impl Uploadable for Placeholder {
+//         fn wasm(_chain: &ChainInfoOwned) -> WasmPath {
+//             // having &self.wasm_path instead would solve the issue
+//             wasm_path
+//         }
+//     }
+//     Placeholder { wasm_path }
+// }
```

Comment:
Yeah, in this update, we have removed `self` from the Uploadable trait. It might have been a bad Idea. 
My proposition to retrieve the self object would be to rollback the change (this will be breaking though) and add a specific method for contract structures to retrieve the wasm path without self (which was the original idea).
@CyberHoward thoughts ?

Reply:
I think the solution that Misha came up with works, so a change isn't "required". 

Yes it's a workaround but this is also a very specific use-case of the library (and not one we really intended it to be used for) so I think this is acceptable.

Reply:
Ok ! Got it

Reply:
> I think the solution that Misha came up with works, so a change isn't "required".
> 
> Yes it's a workaround but this is also a very specific use-case of the library (and not one we really intended it to be used for) so I think this is acceptable.

Same could be said for "very specific use-case" `upload_with_access`, you can just use cosmrs type + commit_tx([]) to upload wasm with access. We don't have to break Uploadable, and implement `upload_wasm(WasmPath)` for `CosmosSender` instead, what do you think?

Reply:
An implementation of this sugestion can be found here : https://github.com/AbstractSDK/cw-orchestrator/pull/487

Reply:
Thanks!